libs/extensions/include/NanoEvent.hpp

```diff
@@ -36,7 +36,11 @@ class NanoEvent {
   float GetNDSAMuon(std::string collectionName);
   float GetNMuon(std::string collectionName);
 
-  std::shared_ptr<PhysicsObject> GetMuonWithIndex(int muon_idx, std::string collectionName, bool isDSAMuon);
+  std::shared_ptr<PhysicsObject> GetMuonWithIndex(int muon_idx, std::string collectionName, bool isDSAMuon=false);
```

Comment:
Actually, here I'm not sure what `isDSAMuon` argument would mean. Does it mean that it will return a DSA muon with this index if I pass `true`? What happens if I pass `false`?

Perhaps this should be broken down into a few methods. So we could have `GetMuonWithIndex` which returns a standard NanoAOD (i.e. PAT) muon, `GetDSAMuonWithIndex` which returns a DSA muon, and a private method like `GetPATorDSAMuonWithIndex(..., bool doDSAmuons=false)` that is called by the other two?

Reply:
Yes I like that idea, and I'll do the same for MuonIndexExists just to avoid the parenthesis bug I had before ;)